src/Language/Haskell/LSP/VFS.hs

```diff
@@ -151,26 +165,27 @@ changeFromServerVFS initVfs (J.RequestMessage _ _ _ params) = do
 
 -- ---------------------------------------------------------------------
 virtualFileName :: FilePath -> J.NormalizedUri -> VirtualFile -> FilePath
-virtualFileName prefix uri (VirtualFile ver _) =
-  prefix </> show (hash (J.fromNormalizedUri uri)) ++ "-" ++ show ver ++ ".hs"
+virtualFileName prefix uri (VirtualFile _ file_ver _) =
+  prefix </> show (hash (J.fromNormalizedUri uri)) ++ "-" ++ show file_ver ++ ".hs"
```

Comment:
I think it would be good to put the original filename in here too, makes it easier to rummage around in the temp directory and identify the file.  Preferably as a prefix.

Reply:
Does `uri` not contain the original file name?

Reply:
Does the uri not contain the original file name?

Reply:
Added.